tools/topology/sof-icl-rt711-rt1308-rt715-hdmi.m4

```diff
@@ -73,7 +73,7 @@ PIPELINE_PCM_ADD(sof/pipe-volume-playback.m4,
 
 # Low Latency capture pipeline 5 on PCM 4 using max 2 channels of s32le.
 # Schedule 48 frames per 1000us deadline on core 0 with priority 0
-PIPELINE_PCM_ADD(sof/pipe-highpass-capture.m4,
+PIPELINE_PCM_ADD(sof/pipe-volume-switch-capture.m4,
```

Comment:
Still not keen on changing the processing here. This was added for a reason, so you may be re-introducing the original problem. @singalsu any comments ?

Reply:
@libinyang please don't change this. We've added a high-pass filter on purpose to avoid DC-related issues.
https://github.com/thesofproject/linux/issues/1480

I still don't get why you would want a mute implemented on the firmware side when it can be done on the RT711 and RT715 side, and the mute can be done in a better way on the 3rd party side (e.g. during a zero crosssing). If we implement the mute in firmware, it should be because it's impossible to do or due to some work-around.

Reply:
@plbossart This is implemented in firmware, not in codec, because this is not a feature in codec. It is a feature in the machine level. Actually the LED feature is non-related with the codec. Even the machine doesn't use codec RT711 and RT715, the LED feature is still there.
@plbossart @lgirdwood Sorry, I don't get the point why you don't want to use volume-switch pipeline.

Reply:
> @plbossart @lgirdwood Sorry, I don't get the point why you don't want to use volume-switch pipeline.

@libinyang I think it is because we need the high-pass filter. Maybe you can add a new switch instead of replacing pipe-highpass-capture.m4?

Reply:
When modifying this pipeline please do a capture test for 10s. There should be no strong start transients. In the SoundWire codec I recall the big "thump" happened for DMIC for the first 2-3 seconds time but please check both DMIC and headset capture.

Reply:
@bardliao @singalsu Thanks for the clarification. We test the volume-switch pipeline and highpass pipeline and both works well. Maybe the "thump" only happens on Intel platform DMIC?

@singalsu For the "thump" noise, Is this because the volume is too high? Maybe we should setting the volume to a proper value instead of the max volume.

Reply:
The "thump" what I called the large DC pulse happened with DMIC that is connected to SoundWire codec, not to DMIC DAI where the issue has been fixed a long time ago. I don't have device to test here but according to other people testing with and without the highpass it helped. If the DC pulse is no more there maybe the codec driver has been altered? See https://github.com/thesofproject/linux/issues/1480

Reply:
@singalsu Thanks for the details.  We tried to test highpass vs volume-capture pipeline. But unfortunately, the highpass pipeline doesn't work now. So do you suggest using highpass with switch pipeline instead of volume-switch pipeline?

Reply:
Can you check if PR #2414 helps. If not please inspect capture start waveform with volume-switch pipeline with every impacted capture PCM. If there are no issues seen then volume-switch usage should be OK. There's no need for high-pass IIR if the problem has been fixed somewhere else (codec driver, codec firmware).

Reply:
@singalsu It seems #2414 cannot help here to fix https://github.com/thesofproject/linux/issues/1816, please see test result of @Liviali155 :https://github.com/thesofproject/linux/issues/1816#issuecomment-590690193

Reply:
@mengdonglin Sorry I missed this. Is the IIR EQ still broken? I will try to test & fix it tomorrow.

Reply:
@singalsu Yvonne has tested both pipe-volume-capture and pipe-highpass-capture. And the result is highpass don't have the noise while pipe-volume-capture still have the noise. I will change to use highpass pipeline. Thanks.

Reply:
@libinyang Good that it was solved!